File: leetcode/problem_448.py
### Problem : 448. Find All Numbers Disappeared in an Array
### tags: Array, Hash Table
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        lookUp = set()
        for num in nums:
            lookUp.add(num)
        
        result = []
        for i in range(1, (len(nums)) + 1):
            if(i not in lookUp):
                result.append(i)
        return result

File: leetcode/problem_448.kt
// Problem 448. Find All Numbers Disappeared in an Array (Easy)
// tags: array, hashtable
// time complexity: O(n)
// space complexity: O(n)
class Solution {
    fun findDisappearedNumbers(nums: IntArray): List<Int> {
        var arr = IntArray(nums.size + 1){0}
        // [4,3,2,7,8,2,3,1]
        // [0, 1, 2, 3, 4, 0, 0, 7, 8]
        for(i in nums){
            arr[i] = i
        }
        var result = mutableListOf<Int>()
        for(i in 1 until arr.size){
            if(arr[i] == 0){
                result.add(i)
            }
        }
        return result
    }
}